### PR TITLE
Lexicon refactor

### PR DESCRIPTION
**Pull request by dholms**
Feature branch for lex refactor
- https://github.com/bluesky-social/atproto/pull/653
- https://github.com/bluesky-social/atproto/pull/655
- https://github.com/bluesky-social/atproto/pull/659
- https://github.com/bluesky-social/atproto/pull/661
- https://github.com/bluesky-social/atproto/pull/664
- https://github.com/bluesky-social/atproto/pull/665
- https://github.com/bluesky-social/atproto/pull/668
- https://github.com/bluesky-social/atproto/pull/673
- https://github.com/bluesky-social/atproto/pull/674
- https://github.com/bluesky-social/atproto/pull/679
- https://github.com/bluesky-social/atproto/pull/680
- https://github.com/bluesky-social/atproto/pull/683
- https://github.com/bluesky-social/atproto/pull/684
- https://github.com/bluesky-social/atproto/pull/685
- https://github.com/bluesky-social/atproto/pull/686
- https://github.com/bluesky-social/atproto/pull/688
- https://github.com/bluesky-social/atproto/pull/690
- https://github.com/bluesky-social/atproto/pull/691
- https://github.com/bluesky-social/atproto/pull/692
- https://github.com/bluesky-social/atproto/pull/693
- https://github.com/bluesky-social/atproto/pull/699
- https://github.com/bluesky-social/atproto/pull/700
- https://github.com/bluesky-social/atproto/pull/707
- https://github.com/bluesky-social/atproto/pull/708
- https://github.com/bluesky-social/atproto/pull/717
- https://github.com/bluesky-social/atproto/pull/718
- https://github.com/bluesky-social/atproto/pull/712
- https://github.com/bluesky-social/atproto/pull/720
- https://github.com/bluesky-social/atproto/pull/723
- https://github.com/bluesky-social/atproto/pull/724
- https://github.com/bluesky-social/atproto/pull/705
- https://github.com/bluesky-social/atproto/pull/727
- https://github.com/bluesky-social/atproto/pull/728
- https://github.com/bluesky-social/atproto/pull/729
- https://github.com/bluesky-social/atproto/pull/731
- https://github.com/bluesky-social/atproto/pull/732
- https://github.com/bluesky-social/atproto/pull/733
- https://github.com/bluesky-social/atproto/pull/735

### CODE REVIEW

File: lexicons/app/bsky/actor/updateProfile.json
Comment (pfrazee) [inline lexicons/app/bsky/actor/updateProfile.json:1]: Did we decide not to remove this yet? I was thinking we'd remove this now that we can do CAS.

File: lexicons/app/bsky/feed/post.json [hunk 1]
```diff
@@ -9,11 +9,16 @@
...
             "type": "array",
+            "description": "Deprecated: replaced by app.bsky.richtext.facet.",
             "items": {"type": "ref", "ref": "#entity"}
           },
+          "facets": {
+            "type": "array",
+            "items": {"type": "ref", "ref": "app.bsky.richtext.facet"}
+          },
           "reply": {"type": "ref", "ref": "#replyRef"},
           "embed": {
```
Comment (pfrazee) [inline lexicons/app/bsky/feed/post.json:23]: @devinivy Are we going to add `images`? (If so make sure to update the postsView in defs.)

File: packages/api/src/client/types/app/bsky/graph/assertCreator.ts
Comment (pfrazee) [inline packages/api/src/client/types/app/bsky/graph/assertCreator.ts:1]: This file should get removed -- codegen problem?

File: packages/api/src/client/types/app/bsky/graph/assertMember.ts
Comment (pfrazee) [inline packages/api/src/client/types/app/bsky/graph/assertMember.ts:1]: This file should get removed -- codegen problem?

File: packages/lexicon/src/types.ts [hunk 2]
```diff
@@ -34,24 +34,33 @@ export const lexInteger = z.object({
 })
 export type LexInteger = z.infer<typeof lexInteger>
 
+export const lexStringFormat = z.enum([
+  'datetime',
+  'uri',
+  'at-uri',
+  'did',
+  'handle',
+  'at-identifier',
```
Comment (pfrazee) [inline packages/lexicon/src/types.ts:43]: Is this "did or handle" ?

**Review by pfrazee — COMMENTED** [review]
A few comments -- I just reviewed the lexicon changes (both the lang updates and the schemas)

**Comment by pfrazee** [comment]
👑 Kings

File: lexicons/app/bsky/actor/updateProfile.json
Comment (devinivy) [reply lexicons/app/bsky/actor/updateProfile.json:1]: This should be removed, I expect being conservative during a merge conflict is what kept this around.

File: lexicons/app/bsky/actor/defs.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        "description": {
+          "type": "string",
+          "maxLength": 256
+        },
+        "avatar": { "type": "string" },
+        "banner": { "type": "string" },
+        "followersCount": {"type": "integer"},
+        "followsCount": {"type": "integer"},
+        "postsCount": {"type": "integer"},
+        "creator": {"type": "string", "format": "did"},
```
[the remark this answers is not in the document]
Comment (devinivy) [reply lexicons/app/bsky/actor/defs.json:39]: Good call, I think we can nix this as part of removing declarations.

File: lexicons/app/bsky/feed/post.json (hunk 1 above)
Comment (devinivy) [reply lexicons/app/bsky/feed/post.json:23]: Yep, I'll plan to hit that today.

File: packages/api/src/client/types/app/bsky/graph/assertCreator.ts
Comment (devinivy) [reply packages/api/src/client/types/app/bsky/graph/assertCreator.ts:1]: Addressed in #707

File: packages/api/src/client/types/app/bsky/graph/assertMember.ts
Comment (devinivy) [reply packages/api/src/client/types/app/bsky/graph/assertMember.ts:1]: Addressed in #707

File: packages/lexicon/src/types.ts (hunk 2 above)
Comment (pfrazee) [reply packages/lexicon/src/types.ts:43]: Works for me

**Pull request merged**